packages/tools/viewer/src/viewer.ts

```diff
@@ -134,7 +137,30 @@ function throwIfAborted(...abortSignals: (Nullable<AbortSignal> | undefined)[]):
     }
 }
 
-function createSkybox(scene: Scene, camera: Camera, reflectionTexture: CubeTexture, blur: number): Mesh {
+async function createCubeTexture(url: string, scene: Scene, extension?: string) {
+    extension = extension ?? GetExtensionFromUrl(url);
+    const instantiateTexture = await (async () => {
+        if (extension === ".hdr") {
+            // eslint-disable-next-line @typescript-eslint/naming-convention
+            const { HDRCubeTexture } = await import("core/Materials/Textures/hdrCubeTexture");
+            return () => new HDRCubeTexture(url, scene, 128, false, true, false, true);
+        } else {
+            // eslint-disable-next-line @typescript-eslint/naming-convention
+            const { CubeTexture } = await import("core/Materials/Textures/cubeTexture");
+            return () => new CubeTexture(url, scene, null, false, null, null, null, undefined, true, extension, true);
+        }
+    })();
+
+    const originalUseDelayedTextureLoading = scene.useDelayedTextureLoading;
+    try {
+        scene.useDelayedTextureLoading = false;

```

Comment:
this needs to be within the promise near the sync part of creating the Texture. it is currently potentially breaking as we download the module in parallel

Reply:
The sync part of creating the texture is the next line (the call to `instantiateTexture`). I'm trying to keep the mutation of `scene.useDelayedTextureLoading` in a synchronous part of the code so it's not possible to have other texture instantiations happen in the window where we mutate this value.

Reply:
I am sure I am missing smthg as the module loading is async and awaited within the instantiateTexture code ? so potentially keeping the delayed state too long ?

Reply:
Nope, the async code imports the module, and then *returns a function* that is used to synchronously instantiate the texture.

Reply:
There is an inline lambda that is invoked and awaited, and the returned value is a synchronous function for creating the texture instance.

Reply:
that is what I was missing, 😴😴😴